File: apps/backend/internal/types/category.go
package types

type CreateCategory struct {
	Name     string `json:"name" binding:"required,min=2,max=20"`
	ParentID string `json:"parentID,omitempty" binding:"omitempty,uuid"`
}

type UpdateCategory struct {
	Name     string `json:"name" binding:"required,min=2,max=20"`
	ParentID string `json:"parentID,omitempty" binding:"omitempty,uuid"`
}

File: packages/types/src/category.ts
import z from "zod";
import { validationErrorResponseSchema } from "./base";

export const createCategorySchema = z.object({
  name: z.string().min(2).max(20),
  parentID: z.uuid().optional(),
});

export const updateCategorySchema = z.object({
  name: z.string().min(2).max(20),
  parentID: z.uuid().optional(),
});

export const createCategoryResponseSchema = validationErrorResponseSchema;
export const updateCategoryResponseSchema = validationErrorResponseSchema;

export type CreateCategory = z.infer<typeof createCategorySchema>;
export type UpdateCategory = z.infer<typeof updateCategorySchema>;

export type CreateCategoryResponse = z.infer<
  typeof createCategoryResponseSchema
>;
export type UpdateCategoryResponse = z.infer<
  typeof updateCategoryResponseSchema
>;
